yb-voyager/cmd/miginfo.go

```diff
@@ -0,0 +1,60 @@
+package cmd
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type MigInfo struct {
+	SourceDBType    string
+	SourceDBName    string
+	SourceDBSchema  string
+	SourceDBVersion string
+	ExportDir       string
+}
+
+func SaveMigInfo(miginfo *MigInfo) error {
+
+	file, err := json.MarshalIndent(miginfo, "", " ")
+	if err != nil {
+		// unable to parse meta info data into json
+		return err
+	}
+
+	migInfoFilePath := filepath.Join(miginfo.ExportDir, META_INFO_DIR_NAME, "miginfo.json")
+
+	err = ioutil.WriteFile(migInfoFilePath, file, 0644)
+	if err != nil {
+		// unable to write miginfo.json file
+		return err
+	}
+
+	return nil
+}
+
+func LoadMigInfo(exportDir string) (*MigInfo, error) {
+
+	metaInfo := &MigInfo{}
```

Comment:
Consider a case where a user renames the export dir *after* the `export schema` is done. This will invalidate the value of `ExportDir` stored in the miginfo.json.

I suggest let's not put `ExportDir` in the serialised json string. We can make the field private with the name `exportDir`. Somewhere around line 57, you can set the field using the value of `exportDir` argument passed to the `LoadMigInfo()` function e.g.

    miginfo.exportDir = exportdir

Reply:
Okay done